lib/evmone/instructions.hpp

```diff
@@ -627,38 +796,51 @@ inline Result returndatacopy(StackTop stack, int64_t gas_left, ExecutionState& s
             return {EVMC_OUT_OF_GAS, gas_left};
 
         if (s > 0)
+        {
             std::memcpy(&state.memory[dst], &state.return_data[src], s);
-    }
 
+            assert(state.requirements != nullptr);
+            if (!std::holds_alternative<Pure>(*size.sval))
+                state.requirements->push_back(Greater{size.sval, 0});
+            auto sreturn_data_offset = std::make_shared<SymbolicMemory>(Offset {input_index.sval, size.sval}, state.sreturn_data);
+            state.smemory = std::make_shared<SymbolicMemory>(SetMem {dst, s, sreturn_data_offset}, state.smemory);
+        }
+    }
     return {EVMC_SUCCESS, gas_left};
 }
 
 inline Result extcodehash(StackTop stack, int64_t gas_left, ExecutionState& state) noexcept
 {
-    auto& x = stack.top();
-    const auto addr = intx::be::trunc<evmc::address>(x);
+    auto& x = stack[0];
+    const auto addr = intx::be::trunc<evmc::address>(x.val);
 
     if (state.rev >= EVMC_BERLIN && state.host.access_account(addr) == EVMC_ACCESS_COLD)
     {
         if ((gas_left -= instr::additional_cold_account_access_cost) < 0)
             return {EVMC_OUT_OF_GAS, gas_left};
     }
 
-    x = intx::be::load<uint256>(state.host.get_code_hash(addr));
+    if (!std::holds_alternative<Pure>(*x.sval))
+        state.requirements->push_back(Equal{x.sval, x.val});
```

Comment:
I think we need to move the requirement up here as well, because it is required for the above gas calculation